web/eslint.config.js

```diff
@@ -1,14 +1,60 @@
+import js from '@eslint/js'
+import pluginReact from 'eslint-plugin-react'
+import pluginPrettier from 'eslint-plugin-prettier'
+import pluginTypescriptEslint from '@typescript-eslint/eslint-plugin'
+import parserTypescriptEslint from '@typescript-eslint/parser'
+import { globalIgnores } from 'eslint/config'
+
 export default [
-    {
-        files: [
-          "./src/**/*.js",
-          "./src/**/*.jsx",
-          "./src/**/*.ts",
-          "./src/**/*.tsx"
-	],
-	extends: [
-          "react-app",
-          "react-app/jest"
-	]
+  globalIgnores(['vite-env.d.ts', 'vite.config.ts', 'dist/**/*']),
+  js.configs.recommended,
+  {
+    files: ['*.ts', '*.tsx'],
+    languageOptions: {
+      parser: parserTypescriptEslint,
+      parserOptions: {
+        project: './tsconfig.json'
+      }
+    },
+    plugins: {
+      '@typescript-eslint': pluginTypescriptEslint
+    },
+    rules: {
+      ...pluginTypescriptEslint.configs['recommended'].rules,
+      ...pluginTypescriptEslint.configs['recommended-requiring-type-checking']
+        .rules,
+      '@typescript-eslint/space-before-function-paren': 'off',
+      '@typescript-eslint/no-extra-semi': 'off'
     }
-];
+  },
+  {
+    files: ['src/react-app-env.d.ts'],
+    rules: {
+      '@typescript-eslint/triple-slash-reference': 'off'
+    }
+  },
```

Comment:
In [react-app-env.d.ts](https://github.com/docat-org/docat/blob/main/web/src/react-app-env.d.ts), did you try to replace
```
/// <reference types="react-scripts" />
```
with
```
import 'react-scripts'
```
?

If that works (and it should accoring to [this guy](https://stackoverflow.com/a/75354018)), this section can be removed

More info on the triple-slash-directive [here](https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html)

Reply:
changed and removed from eslint